clients/codelist/codelist_test.go

```diff
@@ -0,0 +1,1358 @@
+				So(req.Body, ShouldBeNil)
+				So(req.Header.Get(common.AuthHeaderKey), ShouldEqual, common.BearerPrefix+testServiceAuthToken)
+			})
+
+			Convey("and the response body is closed", func() {
+				So(body.IsClosed, ShouldBeTrue)
+			})
+		})
+	})
+}
+
+func TestClient_GetGeographyCodeLists(t *testing.T) {
+	uri := "/code-lists"
+	host := "localhost:8080"
+	query := "type=geography"
+
+	Convey("given clienter.Do returns an error", t, func() {
+		expectedErr := errors.New("master master obey your master")
+		clienter := getClienterMock(nil, expectedErr)
+
+		codelistclient := &Client{
+			url: testHost,
+			cli: clienter,
+		}
+
+		Convey("when codelistClient.GetGeographyCodeLists is called", func() {
+			actual, err := codelistclient.GetGeographyCodeLists(nil, testServiceAuthToken)
+
+			Convey("then the expected error is returned", func() {
+				So(actual, ShouldResemble, CodeListResults{})
+				So(err, ShouldResemble, expectedErr)
+			})
+
+			Convey("and client.Do should be called 1 time with the expected parameters", func() {
+				calls := clienter.DoCalls()
+				So(calls, ShouldHaveLength, 1)
+
+				req := calls[0].Req
+				So(req.URL.Path, ShouldEqual, uri)
+				So(req.URL.RawQuery, ShouldEqual, query)
+				So(req.URL.Host, ShouldEqual, host)
+				So(req.Method, ShouldEqual, "GET")
+				So(req.Body, ShouldBeNil)
+				So(req.Header.Get(common.AuthHeaderKey), ShouldEqual, common.BearerPrefix+testServiceAuthToken)
+			})
+		})
+	})
+
+	Convey("given clienter.Do returns a non 200 response status", t, func() {
+		body := httpmocks.NewReadCloserMock([]byte{}, nil)
+		resp := httpmocks.NewResponseMock(body, 500)
+		clienter := getClienterMock(resp, nil)
+
+		codelistclient := &Client{
+			url: testHost,
+			cli: clienter,
+		}
+
+		Convey("when codelistClient.GetGeographyCodeLists is called", func() {
+			actual, err := codelistclient.GetGeographyCodeLists(nil, testServiceAuthToken)
+
+			Convey("then the expected error is returned", func() {
+				So(actual, ShouldResemble, CodeListResults{})
+
+				expectedErr := &ErrInvalidCodelistAPIResponse{
+					expectedCode: http.StatusOK,
+					actualCode:   500,
+					uri:          testHost + uri + "?" + query,
+				}
+
+				So(err, ShouldResemble, expectedErr)
+			})
+
+			Convey("and client.Do should be called 1 time with the expected parameters", func() {
+				calls := clienter.DoCalls()
+				So(calls, ShouldHaveLength, 1)
+
+				req := calls[0].Req
+				So(req.URL.Path, ShouldEqual, uri)
+				So(req.URL.RawQuery, ShouldEqual, query)
+				So(req.URL.Host, ShouldEqual, host)
+				So(req.Method, ShouldEqual, "GET")
+				So(req.Body, ShouldBeNil)
+				So(req.Header.Get(common.AuthHeaderKey), ShouldEqual, common.BearerPrefix+testServiceAuthToken)
+			})
+
+			Convey("and the response body is closed", func() {
+				So(body.IsClosed, ShouldBeTrue)
+			})
+		})
+	})
+
+	Convey("given ioutil.ReadAll returns an error", t, func() {
+		expectedErr := errors.New("Peace sells, but who's buying?")
```

Comment:
I think convention in GO's error package is to not use capitals or punctuation as per https://github.com/golang/go/wiki/CodeReviewComments#error-strings
Found on lines:
1. 477
2. 597
3. 670
4. 787
5. 858
6. 975
7. 1166
3. 1166